src/lib.rs

```diff
@@ -281,30 +281,52 @@ impl Storage for S3Storage {
             .map_err(|e| zerror!("Get operation failed: {e}"))?
             .map_err(|e| zerror!("Get operation failed: {e}"))?;
 
-        let futures = objects.into_iter().map(|object| {
+        let path_prefix = self.config.path_prefix.to_owned();
```

Comment:
question: Does `config.path_prefix` need to be an owned string or could it be a `&str` here?

Reply:
We can't as it is right now, maybe we could refactor the S3Key struct to contain a &Option<String> for the prefix, but I wouldn't do it for the time being.

Reply:
This seems like a reasonable change with a limited scope. I don't think the lifetime will contaminate much.
Is there something I am missing?

Reply:
Ok I addressed the comment. As a thought, there are other elements in this codebase that can be converted into references, it'll be worth a small refactor in the future.